src/chain/mod.rs

```diff
@@ -113,149 +458,545 @@ async fn get_nonce(port: u16, client: &Client, address: &str) -> Result<u64> {
+        "0x27244d1e0000000000000000000000000000000000000000000000000000000000000000";
+    let zeroth_tba_result =
+        call_contract(port, &addresses.hypermap_proxy, tba_of_zero_calldata).await?;
+    info!("zeroth_tba_result: {}", zeroth_tba_result);
+
+    // Extract address from result (last 20 bytes / 40 hex chars)
+    let zeroth_tba = if zeroth_tba_result.len() >= 42 {
+        format!("0x{}", &zeroth_tba_result[zeroth_tba_result.len() - 40..])
+    } else {
+        return Err(eyre!("Hypermap is not initialized"));
+    };
 
+    info!("Resolved zeroth_tba from hypermap: {}", zeroth_tba);
+    addresses.zeroth_tba = Some(zeroth_tba.clone());
+
+    let client = Client::new();
+    let _impersonator = AnvilImpersonator::new(port, &client, OWNER_ADDRESS).await?;
+    let nonce = get_nonce(port, &client, OWNER_ADDRESS).await?;
+
+    // Build mint calldata: mint(address to, bytes label, bytes initialization, address implementation)
+    let label_hex = "0x6f73"; // "os" label (2 bytes)
+
+    let mint_args = vec![
+        ConstructorArg {
+            arg_type: "address".to_string(),
+            value: OWNER_ADDRESS.to_string(),
+        },
+        ConstructorArg {
+            arg_type: "bytes".to_string(),
+            value: label_hex.to_string(),
+        },
+        ConstructorArg {
+            arg_type: "bytes".to_string(),
+            value: "0x".to_string(),
+        },
+        ConstructorArg {
+            arg_type: "address".to_string(),
+            value: permissioned_minter.clone(),
+        },
+    ];
+
+    let deployed_map = HashMap::new();
+    let mint_calldata = encode_function_call(
+        "mint(address,bytes,bytes,address)",
+        &mint_args,
+        &deployed_map,
+    )?;
+
+    info!("Mint calldata: {}", mint_calldata);
+
+    // Build execute calldata: execute(address,uint256,bytes,uint8)
+    let execute_args = vec![
+        ConstructorArg {
+            arg_type: "address".to_string(),
+            value: addresses.hypermap_proxy.clone(),
+        },
+        ConstructorArg {
+            arg_type: "uint256".to_string(),
+            value: "0".to_string(),
+        },
+        ConstructorArg {
+            arg_type: "bytes".to_string(),
+            value: mint_calldata.clone(),
+        },
+        ConstructorArg {
+            arg_type: "uint8".to_string(),
+            value: "0".to_string(),
+        },
+    ];
+
+    let execute_calldata = encode_function_call(
+        "execute(address,uint256,bytes,uint8)",
+        &execute_args,
+        &deployed_map,
+    )?;
+
+    info!("Execute calldata: {}", execute_calldata);
+
+    // Send transaction to zeroth_tba
+    match execute_transaction(
+        port,
+        &client,
+        OWNER_ADDRESS,
+        Some(&zeroth_tba),
+        &execute_calldata,
+        nonce,
+    )
+    .await
+    {
+        Ok(tx_hash) => {
+            info!("Mint NFT transaction sent: {}", tx_hash);
+
+            sleep(Duration::from_millis(200)).await;
```

Comment:
We need a more robust solution here. Ideally we would have a way to tell that we are ready to proceed by querying the chain. Worst case, we should sleep-retry until ready